source/util/CommandLine.cpp

```diff
@@ -397,6 +413,34 @@ bool CommandLine::parse(span<const string_view> args, ParseOptions options) {
             continue;
         }
 
+        // check if arg is in the list of commands to skip
+        auto cmd_ignore_option = findOption("cmd_ignore");
+        std::map<std::string, int>* cmdIgnore =
+            std::get<std::map<std::string, int>*>(cmd_ignore_option->storage);
+        string_view ignore_arg = arg;
+        // if we ignore a vendor command of the form +xx ,
+        // we match on any +xx+yyy command as +yy is the command's argument
+        if (arg[0] == '+') {
+            size_t plusIndex = arg.substr(1).find_first_of('+');
+            if (plusIndex != string_view::npos)
+                ignore_arg =
+                    arg.substr(0, plusIndex + 1); // +1 because we started from arg.substr(1)
+        }
+        if (auto it{ cmdIgnore->find(std::string(ignore_arg)) }; it != cmdIgnore->end()) {
```

Comment:
I would generally just use the normal `=` initialization for simple variables; the braces aren't buying you anything.

Reply:
I have to admit I took this construct straight out of stackoverflow :-) There are just too many nuances with the different C++ initializer types (direct vs copy vs value), and I didn't want to introduce any subtle bug.
I could also break it into separate lines for assigning the iterator and for testing its value, but the current form keeps the iterator scope inside the `if` block, and I could reuse the short temporary `it` variable.

Reply:
No, using the if statement initialization syntax is totally fine. Mostly I'm just saying to use the straightforward equals sign unless there's a good reason to do otherwise; yes C++ has made a hash of initialization syntax but for the most part the straightforward thing will do what you want so don't worry too much about the braced initialization stuff.